src/content-tags-drawer/data/apiHooks.jsx

```diff
@@ -109,3 +114,13 @@ export const useContentDataResponse = (contentId) => {
 export const useIsContentDataLoaded = (contentId) => (
   useContentData(contentId).status === 'success'
 );
+
```

Comment:
In `useContentData`, the query key is wrong. It needs to include the contentId, so a good query key would be `['contentData', contentId]`. Otherwise the data about XBlock B will be read from the cache of XBlock A and users will see the wrong data.

Same for `useContentTaxonomyTagsData` - the query key needs to include the content ID.

Reply:
I see, makes sense, I updated it. https://github.com/openedx/frontend-app-course-authoring/pull/704/commits/ec2c842656454f958b5af394fd13754cb3e4bbc8